pycbc/vetoes/chisq_cpu.pyx

```diff
@@ -0,0 +1,145 @@
+import numpy
+cimport numpy
+from libc.math cimport cos, sin # This imports c's sin and cos function from the math library
+from cython import wraparound, boundscheck, cdivision
+from pycbc.types import real_same_precision_as
+
+ctypedef fused REALTYPE:
+    float
+    double
+
+ctypedef fused COMPLEXTYPE:
+    float complex
+    double complex
+
+@boundscheck(False)
+@wraparound(False)
+def chisq_accum_bin_cython(numpy.ndarray[REALTYPE, ndim=1] chisq,
+                           numpy.ndarray[COMPLEXTYPE, ndim=1] q, int N):
+    # This can be made parallelized?!
+    for i in range(N):
+        chisq[i] += q[i].real*q[i].real+q[i].imag*q[i].imag
+
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def point_chisq_code(numpy.ndarray[REALTYPE, ndim=1] chisq,
+                     numpy.ndarray[COMPLEXTYPE, ndim=1] v1,
+                     int n, int slen,
+                     numpy.ndarray[REALTYPE, ndim=1] shifts,
+                     numpy.ndarray[numpy.uint32_t, ndim=1] bins,
+                     int blen):
+    # Do I need to declare UINT vs INT??
+    cdef int num_parallel_regions, bstart, bend, i, j, k, r, start, end
+    cdef numpy.ndarray[REALTYPE, ndim=1] outr, outi, pr, pi, vsr, vsi, outr_tmp, outi_tmp
+    cdef COMPLEXTYPE v
+    cdef REALTYPE vr, vi, t1, t2, k1, k2, k3, vs, va
+
+    #real_type = real_same_precision_as(chisq)
+    real_type = numpy.float64
```

Comment:
Did we do this all in float64 before? Does that have performance implications?

Reply:
No, this type depended on the input type. It will here too, but I haven't included that yet while I tested performance between weave and this code at double precision.

Reply:
This is now removed as I replaced `numpy.zeros` with C mallocs.